### ISSUE TITLE
when I load pre-trained model of version transformer and CSPNet, it appears mismatch error of gfl_cls parameters

### ISSUE THREAD

**Issue by xxy90** [RangiLyu/nanodet#231]
![image](https://user-images.githubusercontent.com/32832726/116201692-5d62e180-a76c-11eb-91b5-d64034b4c3f9.png)

**Comment by RangiLyu** [comment]
Make sure the num_classes in your config file is the same with the pre-trained model.

**Comment by ag-ds-bubble** [comment]
Hello, i have the same query, do you mean to say that even for out custom dataset which lets say have 4 target classes to predict, we have to still keep the num_classes same as the pre-trained model?

Actually i wanted to do transfer learning, which would mean replacing the output classes to 4. But when i do that using the config file like this.. 
`
#Config File example
save_dir: ../../results/detector/nanodet/nanodet-m-tl
model:
  arch:
    name: OneStageDetector
    backbone:
      name: ShuffleNetV2
      model_size: 1.0x
      out_stages: [2,3,4]
      activation: LeakyReLU
    fpn:
      name: PAN
      in_channels: [116, 232, 464]
      out_channels: 96
      start_level: 0
      num_outs: 3
    head:
      name: NanoDetHead
      num_classes: 4
      input_channel: 96
      feat_channels: 96
      stacked_convs: 2
      share_cls_reg: True
      octave_base_scale: 5
      scales_per_octave: 1
      strides: [8, 16, 32]
      reg_max: 7
      norm_cfg:
        type: BN
      loss:
        loss_qfl:
          name: QualityFocalLoss
          use_sigmoid: True
          beta: 2.0
          loss_weight: 1.0
        loss_dfl:
          name: DistributionFocalLoss
          loss_weight: 0.25
        loss_bbox:
          name: GIoULoss
          loss_weight: 2.0
data:
  train:
    name: coco
    img_path: ../../raw_data/__temp__/common_images/
    ann_path: ../../prepared_data/detector/nanodet/train.json
    input_size: [320,320] #[w,h]
    keep_ratio: True
    pipeline:
      perspective: 0.3
      rotation: 180
      flip: 0.3
      normalize: [[103.53, 116.28, 123.675], [57.375, 57.12, 58.395]]
  val:
    name: coco
    img_path: ../../raw_data/__temp__/common_images/
    ann_path: ../../prepared_data/detector/nanodet/val.json
    input_size: [320,320] #[w,h]
    keep_ratio: True
    pipeline:
      perspective: 0.3
      rotation: 180
      flip: 0.3
      normalize: [[ 103.53, 116.28, 123.675 ], [ 57.375, 57.12, 58.395 ]]
device:
  gpu_ids: [0]
  workers_per_gpu: 16
  batchsize_per_gpu: 80
schedule:
  resume: True
  load_model: ../../pretrained_weights/detector/nanodet/nanodet_m.ckpt
.....
`
nanodet_m.ckpt taken from [here](https://drive.google.com/file/d/1ZkYucuLusJrCb_i63Lid0kYyyLvEiGN3/view?usp=sharing)

How should i go about replacing the final layer from 80 classes which is the default to 4?

**Comment by RangiLyu** [comment]
fixed in ad66491

**Comment by chuaziheng** [comment]
Hi , I have a similar problem, if i am using a pretrained model trained on COCO dataset (80 classes), but i have only 3 target classes, what shoud I put for the model.arch.head.num_classes and class_names? Thank you

**Comment by ag-ds-bubble** [comment]
I am still facing the same issue, could you kindly provide an example to use nanodet for trasfer learning where the `num_classes` differ from the classes in `load_model`?
